### PR TITLE
Allow `npm run build` to work cross-platform.

### PR DESCRIPTION
**Pull request by sysrage**
This allows users to simply run `npm run build` no matter what platform they're using. OSX/Linux users no longer need to run `npm run build:nix`.

I have only tested this in OSX, but it should still work properly in Windows.

Thanks you Saddie for the platform detection.

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/CUModSquad/cu-patcher-ui/pull/30" target="_blank">ReviewNinja

File: package.json [hunk 1]
```diff
@@ -35,18 +36,13 @@
...
-    "copy": "npm run copy:third-party && npm run copy:misc",
-    "copy:nix": "npm run copy:third-party:nix && npm run copy:misc:nix",
+    "copy": "npm run is-windows --loglevel=silent && (npm run copy:third-party && npm run copy:misc) || (npm run copy:third-party:nix && npm run copy:misc:nix)",
     "babel": "babel tmp -d tmpp",
-    "browserify": "browserify tmpp/main.js -o dist/js/%npm_package_config_name%.js",
-    "browserify:nix": "browserify tmpp/main.js -o dist/js/$npm_package_config_name.js",
-    "prebrowserify": "mkdirp dist/js",
-    "prebrowserify:nix": "mkdir -p dist/js",
+    "browserify": "npm run is-windows --loglevel=silent && browserify tmpp/main.js -o dist/js/%npm_package_config_name%.js || browserify tmpp/main.js -o dist/js/$npm_package_config_name.js",
+    "prebrowserify": "npm run is-windows --loglevel=silent && mkdirp dist/js || mkdir -p dist/js",
```
Comment (Mehuge) [inline package.json]: There is an issue here.  If mkdirp dist/js failes, it will also run mkdir -p dist/js.

File: package.json [hunk 2]
```diff
@@ -35,18 +36,13 @@
...
     "copy:third-party:nix": "cp -av src/third-party dist",
-    "copy": "npm run copy:third-party && npm run copy:misc",
-    "copy:nix": "npm run copy:third-party:nix && npm run copy:misc:nix",
+    "copy": "npm run is-windows --loglevel=silent && (npm run copy:third-party && npm run copy:misc) || (npm run copy:third-party:nix && npm run copy:misc:nix)",
     "babel": "babel tmp -d tmpp",
-    "browserify": "browserify tmpp/main.js -o dist/js/%npm_package_config_name%.js",
-    "browserify:nix": "browserify tmpp/main.js -o dist/js/$npm_package_config_name.js",
-    "prebrowserify": "mkdirp dist/js",
-    "prebrowserify:nix": "mkdir -p dist/js",
+    "browserify": "npm run is-windows --loglevel=silent && browserify tmpp/main.js -o dist/js/%npm_package_config_name%.js || browserify tmpp/main.js -o dist/js/$npm_package_config_name.js",
```
Comment (Mehuge) [inline package.json]: There is an issue here. If browserify failes, it will also run the nix version of browserify

**Comment by Mehuge** [comment]
Consider this:
  some-test && command-a || command-b
if command a fails, command b will be executed
Examples:-
  true && false || echo hello
outputs hello
  true && { echo ok; false; } || { echo fail; }
outputs ok and fail

**Comment by Mehuge** [comment]
Feels to me like what is needed is some way to select the top level script (build:win or build:nix) and then just have those reference their win/nix sub-tasks rather that repeatedly testing windows/nix.

It's a shame can't do something like "build": "npm run build:`node -e 'executecodetogetplatform'`"

**Comment by sysrage** [comment]
Closing. Need a better solution besides && / ||.

**Pull request closed without merging**